webpack/scenes/Subscriptions/Manifest/ManageManifestModal.js

```diff
@@ -57,12 +64,46 @@ class ManageManifestModal extends Component {
     }
   };
 
-  updateRepositoryUrl = (event) => {
-    this.setState({ redhat_repository_url: event.target.value });
+  updateCdnUrl = (event) => {
+    this.setState({ cdn_url: event.target.value });
+  };
+
+  updateCdnUsername = (event) => {
+    this.setState({ cdn_username: event.target.value });
+  };
+
+  hidePasswordPlaceholder = (event) => {
+    const { target } = event;
+    target.value = this.state.cdn_password;
```

Comment:
Mutating the value of a DOM element controlled by React in this way can cause unexpected behavior, since React is the one that decides when it wants to render or update (or unmount) the element. I prefer making it a [controlled component](https://reactjs.org/docs/forms.html#controlled-components), in other words storing that FormControl's value in state (by giving `<FormControl>` a `value` prop) and updating the state instead.

Reply:
That makes sense. I'll play around with it